test/odbc/psqlodbc/psqlodbc_tests_common.h

```diff
@@ -0,0 +1,311 @@
+
+void insertValuesInTable(ServerType serverType, const string &table_name, const vector<string> &insertedValues, bool isNumericInsert);
+
+void insertValuesInTable(ServerType serverType, const string &tableName, const string &insertString, int numRows);
+
+void verifyValuesInObject(ServerType serverType, const string &objectName, const string &orderByColumnName, 
+  const vector<string> &insertedValues, const vector<string> &expectedInsertedValues);
+
+void testCommonColumnAttributes(ServerType serverType, const string &tableName, int numCols, const string &orderByColumnName, 
+  const vector<int> &lengthExpected, const vector<int> &precisionExpected, const vector<int> &scaleExpected, const vector<string> &nameExpected);
+
+void testCommonCharColumnAttributes(ServerType serverType, const string &tableName, int numCols, const string &orderByColumnName, 
+  const vector<int> &lengthExpected, const vector<int> &precisionExpected, const vector<int> &scaleExpected, const vector<string> &nameExpected, 
+  const vector<int> &caseSensitivityExpected, const vector<string> &prefixExpected, const vector<string> &suffixExpected);
+
+void testTableCreationFailure(ServerType serverType, const string &tableName, const vector<vector<pair<string, string>>> &invalidColumns);
+
+void testInsertionSuccess(ServerType serverType, const string &tableName, const string &orderByColumnName, 
+  const vector<string> &insertedValues, const vector<string> &expectedInsertedValues);
+
+void testInsertionFailure(ServerType serverType, const string &tableName, const string &orderByColumnName, 
+  const vector<string> &invalidInsertedValues, bool isNumericInsert);
+
+void testUpdateSuccess(ServerType serverType, const string &tableName, const string &orderByColumnName, 
+  const string &colNameToUpdate, const vector<string> &updatedValues, const vector<string> &expectedUpdatedValues);
+
+void testUpdateFail(ServerType serverType, const string &tableName, const string &orderByColumnName, 
+  const string &colNameToUpdate, const vector<string> &expectedInsertedValues, const vector<string> &updatedValues);
+
+void testPrimaryKeys(ServerType serverType, const string &schemaName, const string &pkTableName, const vector<string> &primaryKeyColumns);
+
+void testUniqueConstraint(ServerType serverType, const string &tableName, const vector<string> &uniqueConstraintColumns);
+
+void testComparisonOperators(ServerType serverType, const string &tableName, const string &col1Name, const string &col2Name, 
+  const vector<string> &col1Data, const vector<string> &col2Data, const vector<string> &operationsQuery, const vector<vector<char>> &expectedResults);
+
+void testComparisonFunctions(ServerType serverType, const string &tableName, const vector<string> &operationsQuery, const vector<string> &expectedResults);
+
+template <typename T>
+void verifyValuesInObject(ServerType serverType, string objectName, string orderByColumnName, int type, T data, 
```

Comment:
Could you declare the templated headers with the rest of them to separate the implementation and the headers?
ie
Having the two `verifyValuesInObject` next to each other (as just header) and the implementation still down here

Reply:
Done.